lib/galaxy/schema/jobs.py

```diff
@@ -0,0 +1,233 @@
+import json
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+)
+
+from pydantic import (
+    Extra,
+    Field,
+    Required,
+    UUID4,
+    validator,
+)
+from typing_extensions import Literal
+
+from galaxy.schema.fields import (
+    DecodedDatabaseIdField,
+    EncodedDatabaseIdField,
+)
+from galaxy.schema.schema import (
+    DataItemSourceType,
+    EncodedDataItemSourceId,
+    EntityIdField,
+    JobState,
+    JobSummary,
+    Model,
+)
+
+
+class JobInputSummary(Model):
+    has_empty_inputs: bool = Field(
+        default=Required,
+        title="Empty inputs",
+        description="Job has empty inputs.",
+    )
+    has_duplicate_inputs: bool = Field(
+        default=Required,
+        title="Duplicate inputs",
+        description="Job has duplicate inputs.",
+    )
+
+
+# TODO: Use Tuple again when `make update-client-api-schema` supports them
+class JobErrorSummary(Model):
+    # messages: List[Union[Tuple[str, str], List[str]]]
+    messages: List[List[str]] = Field(
+        default=Required,
+        title="Error messages",
+        description="The error messages for the specified job.",
+    )
+
+
+class JobAssociation(Model):
+    name: str = Field(
+        default=Required,
+        title="name",
+        description="Name of the job parameter.",
+    )
+    dataset: EncodedDataItemSourceId = Field(
+        default=Required,
+        title="dataset",
+        description="Reference to the associated item.",
+    )
+
+
+class JobInputAssociation(JobAssociation):
+    name: str = Field(
+        default=Required,
+        title="name",
+        description="Name of the job input parameter.",
+    )
+
+
+class JobOutputAssociation(JobAssociation):
+    name: str = Field(
+        default=Required,
+        title="name",
+        description="Name of the job output parameter.",
+    )
+
+
+class ReportJobErrorPayload(Model):
+    dataset_id: DecodedDatabaseIdField = Field(
+        default=Required,
+        title="History Dataset Association ID",
+        description="The History Dataset Association ID related to the error.",
+    )
+    email: Optional[str] = Field(
+        default=None,
+        title="Email",
+        description="Email address for communication with the user. Only required for anonymous users.",
+    )
+    message: Optional[str] = Field(
+        default=None,
+        title="Message",
+        description="The optional message sent with the error report.",
+    )
+
+
+class SearchJobsPayload(Model):
+    tool_id: str = Field(
+        default=Required,
+        title="Tool ID",
+        description="The tool ID related to the job.",
+    )
+    inputs: Dict[str, Any] = Field(
+        default=Required,
+        title="Inputs",
+        description="The inputs of the job.",
+    )
+    state: JobState = Field(
+        default=Required,
```

Comment:
This was previously optional, and now breaks a BioBlend test: https://github.com/nsoranzo/bioblend/actions/runs/6970136619/job/18968153613#step:10:3258

Can it be made optional again?